pkg/cache/internal/cache_reader.go

```diff
@@ -73,6 +75,21 @@ func (c *CacheReader) Get(_ context.Context, key client.ObjectKey, out runtime.O
 	// TODO(directxman12): revisit the decision to always deepcopy
 	obj = obj.(runtime.Object).DeepCopyObject()
 
+	// If out is supposed to be unstructured handle correctly.
+	if o, ok := out.(*unstructured.Unstructured); ok {
```

Comment:
I'm not sure I'm following the logic behind the marshalling and unmarshalling... shouldn't the object already be an unstructured?  We should just be able to do the reflection hack...